Dev

SimpleFNAFer opened the pull request.

H-b-IO-T-O-H committed:
add files

H-b-IO-T-O-H committed:
add sql sheme

H-b-IO-T-O-H committed:
add sql sheme

H-b-IO-T-O-H committed:
add sql sheme

SimpleFNAFer committed:
Added initScheme.sql

-Created description to 5 tables: Users, Groups, Students, Professors, Competenties

SimpleFNAFer committed:
Edited initScheme.sql

-In table Professors: column "rank" renamed to "prof_rank"

SimpleFNAFer committed:
Edited initScheme.sql

-9 more tables added
-3 more types added

H-b-IO-T-O-H commented on scripts/initScheme.sql:
```diff
@@ -0,0 +1,172 @@
+drop table if exists Days cascade;
+drop table if exists Lessons cascade;
+
+drop type if exists user_roles;
```
users_roles

H-b-IO-T-O-H commented on scripts/initScheme.sql:
```diff
@@ -0,0 +1,172 @@
+	phone 			varchar(18),
+	birth_date 		date,
+	about 			text
+)
```
Здесь и далее везде после описания таблиц нужны ";"

H-b-IO-T-O-H commented on scripts/initScheme.sql:
```diff
@@ -0,0 +1,172 @@
+	work_rate 	int4, 			/*Ставка в руб.*/
+	work_time 	varchar(32), 		/*Время работы на кафедре*/
+	disciplines 	varchar(128)[],
+	foreign key (each element of disciplines) references Disciplines(disc_name)
```
Насколько мне известно, postgresql не поддерживает связи такого вида. Массив из внешних ключей точно не укладывается в реляционную модель, в постгре с этим строго. Хотя, возможно, и будут патчи в будущем.

H-b-IO-T-O-H commented on scripts/initScheme.sql:
```diff
@@ -0,0 +1,172 @@
+(
+	competention 	varchar(64) primary key not null,
+	users_ids 	uuid[],
+	foreign key (each element of users_ids) references Users(user_id)
```
Аналогично с комментом выше

H-b-IO-T-O-H commented on scripts/initScheme.sql:
```diff
@@ -0,0 +1,172 @@
+(
+	group_nmb 	int2 references Groups(group_nmb) not null,
+	semester 	int2 not null,
+	weeks_nmbs 	int2(18)[],
```
Это значит, что в int2 ограничение на 18 байт?)

H-b-IO-T-O-H commented on scripts/initScheme.sql:
```diff
@@ -0,0 +1,172 @@
+
+create unlogged table Professors
+(
+	user_id 	uuid foreign key references Users(user_id) not null,
```
foreign key не нужно писать, references уже это подразумевает

H-b-IO-T-O-H reviewed (changes requested):
Мелкие замечания

SimpleFNAFer committed:
Corrected initScheme.sql, Edited sqlScheme

-Mistakes described in review corrected
-Added postgresql dialect
-Edited column group_nmb in table Groups
